faculty/clients/job.py

```diff
@@ -125,6 +155,113 @@ def make_job_summary(self, data):
         return JobSummary(**data)
 
 
+class InstanceSizeSchema(BaseSchema):
+    milli_cpus = fields.Integer(data_key="milliCpus", required=True)
+    memory_mb = fields.Integer(data_key="memoryMb", required=True)
+
+    @post_load
+    def make_instance_size(self, data):
+        return InstanceSize(**data)
+
+
+class JobParameterSchema(BaseSchema):
+    name = fields.String(required=True)
+    type = EnumField(ParameterType, by_value=True, required=True)
+    default = fields.String(required=True)
+    required = fields.Boolean(required=True)
+
+    @post_load
+    def make_job_parameter(self, data):
+        return JobParameter(**data)
+
+
+class JobCommandSchema(BaseSchema):
+    name = fields.String(required=True)
+    parameters = fields.List(fields.Nested(JobParameterSchema), required=True)
+
+    @post_load
+    def make_job_command(self, data):
+        return JobCommand(**data)
+
+
+class JobDefinitionSchema(BaseSchema):
+    working_dir = fields.String(data_key="workingDir", required=True)
+    command = fields.Nested(JobCommandSchema, required=True)
+    image_type = EnumField(
+        ImageType, by_value=True, data_key="imageType", required=True
+    )
+    conda_environment = fields.String(
+        data_key="condaEnvironment", missing=None
+    )
+    environment_ids = fields.List(
+        fields.String(), data_key="environmentIds", required=True
+    )
+    instance_size_type = fields.String(
+        data_key="instanceSizeType", required=True
+    )
+    instance_size = fields.Nested(InstanceSizeSchema, data_key="instanceSize")
+    max_runtime_seconds = fields.Integer(
+        data_key="maxRuntimeSeconds", required=True
+    )
+
+    @validates_schema
+    def validate_conda_environment(self, data):
+        image_is_r = data["image_type"] == ImageType.R
+        print("HERE")
+        print(data["image_type"])
+        if image_is_r:
+            print("IMAGE IS R")
+        image_is_python = data["image_type"] == ImageType.PYTHON
+        if image_is_python:
+            print("IMAGE IS PYTHON")
+        conda_environment_set = data["conda_environment"] is not None
+        if conda_environment_set:
+            print("CONDA ENVIRONMENT IS SET")
+        if image_is_r and conda_environment_set:
+            raise ValidationError(
+                "conda environment must only be set for python images"
```

Comment:
Minor nit for consistency.
```suggestion
                "conda environment must only be set for Python images"
```